src/Request.js

```diff
@@ -306,13 +319,16 @@ export default class Request {
                 new_path += '?' + data;
             }
         }
-        else if (this.auto_format) {
-            payload = (!this.headers['Content-Type'] && this.data)
-                ? Request.stringify(this.data)
-                : JSON.stringify(this.data);
-        }
-        else {
-            payload = this.data;
+        else if (!this._formatted_payload) {
```

Comment:
With this change we may also want to clearly indicate that each request should NOT be re-used (since once formatted, the data cannot be changed)

I'm also not sure if we ever had a case of re-using requests?

Reply:
The request can still be reused because of this change:
https://github.com/anyTV/cuddle/blob/4792cb0e7c014883e4b7beac25dad605920c8608/src/Request.js#L294-L298

We unset `this._formatted_payload` every time we set `this.data` via `send()`, hence changing the body will trigger the payload auto-formatting again.